### DISCUSSION TITLE
Point inside given shape

### DISCUSSION THREAD

**Discussion by Rennngw** [coal-library/coal#494] [Q&A] [answered]
Hi,

I have a static closed STL shape in 3D (for example, a tetrahedron). I am sampling points randomly in 3D space. I need to consider points that are inside the shape.

Is it possible to implement using the FCL library?

**Comment by lmontaut** [comment, marked as answer]
Hi @Rennngw,
In the `hppfcl3x` branch, if your object is convex, you can load it by using a `MeshLoader` then call the `buildConvexHull` method of BVHs:
```
std::shared_ptr<ConvexBase> loadConvexShape(const std::string& file_name) {
  NODE_TYPE bv_type = BV_AABB;
  MeshLoader loader(bv_type);
  BVHModelPtr_t bvh = loader.load(file_name);
  bvh->buildConvexHull(true, "Qt");
  return bvh->convex;
}

// ...
// inside main function
std::shared_ptr<ConvexBase> shape = loadConvexShape("obj.stl");
```

You can then access the faces normals and faces offsets of the mesh to check if your points are inside/outside the shapes (by considering all hyperplanes defining your shape):
```
shape->normals[i];
shape->offsets[i];
```
